packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1799,6 +1807,11 @@ class SemanticsProperties extends DiagnosticableTree {
   ///  * [CustomSemanticsAction], for an explanation of custom actions.
   final Map<CustomSemanticsAction, VoidCallback>? customSemanticsActions;
 
+  /// {@template flutter.semantics.SemanticsProperties.role}
+  /// A enum to describe what role the subtree represents.
```

Comment:
This could be a little more informative. See https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#writing-prompts-for-good-documentation for prompts.

Questions I have:
* What does it mean to give a subtree a role?
* What are the available roles?
* What do a11y systems do with this?
* What's the difference between leaving this null and setting it to SemanticsRole.none?